<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PIClass</class>
 <widget class="QWidget" name="PIClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>888</width>
    <height>555</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>PI</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_10">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QPushButton" name="StartBtn">
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>设定线程数</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="ThreadnumEd"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>串行程序计算时间</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="serialTimeEd"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>并行程序计算时间</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="paralleltimeEd"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_8">
         <item>
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>串行程序计算结果</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="serialrstEd"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_9">
         <item>
          <widget class="QLabel" name="label_9">
           <property name="text">
            <string>并行程序计算结果</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="paralrstEd"/>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>加速比</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="speedupEd"/>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <spacer name="horizontalSpacer_4">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;蒙特卡罗方法（Monte Carlo method），也称统计模拟方法，&lt;/p&gt;&lt;p&gt;是1940年代中期由于科学技术的发展和电子计算机的发明，&lt;/p&gt;&lt;p&gt;而提出的一种以概率统计理论为指导的数值计算方法。&lt;/p&gt;&lt;p&gt;是指使用随机数（或更常见的伪随机数）来解决很多计算问题的方法。&lt;/p&gt;&lt;p&gt;正方形的面积为：2r * 2r=4r^{2}&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;圆的面积为：pi r ^{2}&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;圆的面积比上正方形的面积为：π / 4&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;所以我们使用蒙特卡洛法在正方形内随机撒点，&lt;/p&gt;&lt;p&gt;落在圆内的点 / 落在正方形内的点（全部的点），&lt;/p&gt;&lt;p&gt;就约等于圆的面积 / 正方形的面积 = π / 4&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
